### PR TITLE
Add new event for application callback when no active jobs are available

### PR DESCRIPTION
**Pull request by ActoryOu**
Add no active job event in application callback

Description
-----------

This change adds a new event in application callback. This provides a callback when no active job is available in service to process. Useful in cases where device needs to shutdown OTA agent if no active jobs are available to save power.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by pvyawaha** [comment]
Actory is working on this PR which has changes from https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/416 so closing 416

**Comment by pvyawaha** [comment]
Actory is working on this PR which has changes from https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/416 so closing 416

**Review by pvyawaha — DISMISSED** [review]

**Review by paulbartell — APPROVED** [review]

File: source/ota.c [hunk 1]
```diff
@@ -2238,6 +2238,10 @@ static void handleJobParsingError( const OtaFileContext_t * pFileContext,
             LogInfo( ( "No active job available in received job document: "
                        "OtaJobParseErr_t=%s",
                        OTA_JobParse_strerror( err ) ) );
+
+            /* Callback when no active job is available to be processed. */
+            otaAgent.OtaAppCallback( OtaJobEventNoActiveJob, NULL );
```
Comment (paulbartell) [inline source/ota.c:2243]: Is OtaAppCallback verified to be non-null anywhere?

**Review by paulbartell — CHANGES_REQUESTED** [review]
OtaAppCallback should checked that it is non-null prior to calling.

File: source/ota.c (hunk 1 above)
Comment (ActoryOu) [reply source/ota.c]: To fix this issue in a common way, I added a callOtaCallback function check if callback function exists before calling OtaAppCallback.

**Review by pvyawaha — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**